examples/chapter1/CMakeLists.txt

```diff
@@ -23,66 +23,117 @@ PROJECT(${PROJ_NAME})
 
 # Include directories with headers
 #---------------------------------------------------#
-INCLUDE_DIRECTORIES( ${CMAKE_CURRENT_SOURCE_DIR}/include )
-INCLUDE_DIRECTORIES( ${KLEPSYDRA_INCLUDE_DIRS} )
-
 ADD_EXECUTABLE(${PROJ_NAME}_1 ${${PROJ_NAME}_HEADERS} src/example1.cpp )
 
 TARGET_LINK_LIBRARIES(${PROJ_NAME}_1
+  PUBLIC
   ${KLEPSYDRA_CORE_LIBRARIES}
-  ${CMAKE_THREAD_LIBS_INIT}
   atomic
   )
 
+TARGET_INCLUDE_DIRECTORIES(${PROJ_NAME}_1 
+  PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/include 
+  ${KLEPSYDRA_INCLUDE_DIRS}
+  )
+
 ADD_EXECUTABLE(${PROJ_NAME}_2 ${${PROJ_NAME}_HEADERS} src/example2.cpp )
 
 TARGET_LINK_LIBRARIES(${PROJ_NAME}_2
+  PUBLIC
   ${KLEPSYDRA_CORE_LIBRARIES}
-  ${CMAKE_THREAD_LIBS_INIT}
   atomic
   )
 
+TARGET_INCLUDE_DIRECTORIES(${PROJ_NAME}_2 
+  PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/include 
+  ${KLEPSYDRA_INCLUDE_DIRS}
```

Comment:
Same comments here as the previous target. You should not need to put klepsydra_include_dirs anywhere.

Reply:
I tried it but it didn't compile too

Reply:
After the kpsr-core update in my PC this works . So, done